pkg/secretless/plugin/sharedobj/available_plugins_test.go

```diff
@@ -6,263 +6,161 @@ import (
+		assert.Equal(t, tcpPlugins, mock.TCPPluginsById)
+	})
 }
 
 func TestAllAvailablePlugins(t *testing.T) {
 	t.Run("Assembles internal and external plugins", func(t *testing.T) {
 		allPlugins, err := AllAvailablePluginsWithOptions(
 			"",
 			"",
-			mockGetInternalPlugins,
-			mockGetExternalPlugins,
-			newLogger(),
+			mock.GetInternalPlugins,
+			mock.GetExternalPlugins,
+			mock.NewLogger(),
 		)
 		assert.Nil(t, err)
 		if err != nil {
 			t.Fail()
 		}
 
-		assert.Equal(t, 5, len(allPlugins.HTTPPlugins()))
-		assert.Equal(t, 6, len(allPlugins.TCPPlugins()))
-
-		assertMapContainsHTTPPlugins(t, allPlugins, mock.InternalPlugins().HTTPPlugins())
-		assertMapContainsHTTPPlugins(t, allPlugins, mockExternalPlugins().HTTPPlugins())
-
-		assertMapContainsTCPPlugins(t, allPlugins, mock.InternalPlugins().TCPPlugins())
-		assertMapContainsTCPPlugins(t, allPlugins, mockExternalPlugins().TCPPlugins())
+		assert.EqualValues(t, mock.AllHTTPPlugins(), allPlugins.HTTPPlugins())
+		assert.EqualValues(t, mock.AllTCPPlugins(), allPlugins.TCPPlugins())
 	})
 
 	t.Run("External plugins override same-named internal plugins", func(t *testing.T) {
-		defaultExternalPlugins, _ := mockGetExternalPlugins("", "", newLogger())
 
-		httpPlugins := defaultExternalPlugins.HTTPPlugins()
-		httpPlugins["intHTTP2"] = mock.HTTPPlugin{}
+		// Setup our mocks
 
-		tcpPlugins := defaultExternalPlugins.TCPPlugins()
-		tcpPlugins["intTCP1"] = mock.TCPPlugin{}
+		extHTTPPlugins := mock.ExternalPlugins().HTTPPlugins()
+		extHTTPPlugins["intHTTP2"] = mock.HTTPPlugin{}
 
-		allExternalPlugins := plugin.Plugins{
-			HTTPPluginsByID: httpPlugins,
-			TCPPluginsByID:  tcpPlugins,
+		extTCPPlugins := mock.ExternalPlugins().TCPPlugins()
+		extTCPPlugins["intTCP1"] = mock.TCPPlugin{}
+
+		allExternalPlugins := Plugins{
+			HTTPPluginsByID: extHTTPPlugins,
+			TCPPluginsByID:  extTCPPlugins,
 		}
 
-		mockGetExternalPluginsWithOverride := func(
+		getExternalPlugins := func(
 			pluginDir string,
 			checksumfile string,
 			logger log_api.Logger,
 		) (plugin.AvailablePlugins, error) {
 			return &allExternalPlugins, nil
 		}
 
-		allInternalPlugins, _ := mockGetInternalPlugins()
-		mockGetInternalPluginsWithOverride := func() (plugin.AvailablePlugins, error) {
-			return allInternalPlugins, nil
-		}
+		// Create the test subject
 
 		allPlugins, err := AllAvailablePluginsWithOptions(
 			"",
 			"",
-			mockGetInternalPluginsWithOverride,
-			mockGetExternalPluginsWithOverride,
-			newLogger(),
+			mock.GetInternalPlugins,
+			getExternalPlugins,
+			mock.NewLogger(),
 		)
 
+		// Test
+
 		assert.Nil(t, err)
 		if err != nil {
 			t.Fail()
 		}
 
-		assert.Equal(t, 5, len(allPlugins.HTTPPlugins()))
-		assert.Equal(t, 6, len(allPlugins.TCPPlugins()))
+		// Ensure the override occurred
+		assert.Equal(t, extHTTPPlugins["intHTTP2"], allPlugins.HTTPPlugins()["intHTTP2"])
+		assert.Equal(t, extTCPPlugins["intTCP1"], allPlugins.TCPPlugins()["intTCP1"])
```

Comment:
Not a blocker: This test is a much weaker assertion - we don't know for a fact that the rest of the plugins are unchanged

Blocker: `assert.Equal` compares values, not pointers so this test isn't testing the right thing afaict since both the original and replacement structs have the same empty value.